Dockerfile-multiarch

```diff
@@ -0,0 +1,19 @@
+# Dockerfile for ghostunnel/ghostunnel, used for multi-arch release builds.
+#
+# To run ghostunnel from the image (for example):
+#     docker run --rm ghostunnel/ghostunnel --version
+
+# Work out which binary in dist/ to use
+FROM debian:buster-slim as build
+ARG TARGETPLATFORM
+ARG VERSION
+ADD dist/ dist/
+RUN ARCH=$(echo ${TARGETPLATFORM} | cut -d'/' -f2) && \
+    cp dist/ghostunnel-${VERSION}-linux-${ARCH}-with-pkcs11 /usr/bin/ghostunnel
```

Comment:
I think this could possibly just be done with:
```
ARG ARCH
ARG VERSION
ADD dist/ghostunnel-${VERSION}-linux-${ARCH}-with-pkcs11 /usr/bin/ghostunnel
```

In which case we could also skip the multi-stage part of this Dockerfile (assuming that works)

Reply:
The reason behind this is because docker buildx provides a TARGETPLATFORM arg which is populated with something like `linux/arm64`. This is to convert that to just `arm64` so we can use it to pick the binary. However if we're going to switch back to using Alpine, where we'll have to compile in the build again this can all be deleted.

Reply:
> The reason behind this is because docker buildx provides a TARGETPLATFORM arg...

You can also use the provided `TARGETARCH` which skips the `linux/` part and just gives you `amd64`, `arm64` etc.
[Example](https://gitlab.com/jitesoft/dockerfiles/php/-/blob/master/Dockerfile#L32)
Just remember that it MUST be declared before  being used, it's provided only if you add the `ARG TARGETARCH` part.

Reply:
Oh that's cool. Thanks! Doesn't seem to be documented anywhere :laughing: In any case the way @csstaub suggested doing things does drastically simplify things so we don't need any of this complexity, even if builds take a little longer.

Reply:
Aye, it's quite annoying, I have found quite a bit of information by reading through the code trying to figure out how it works! ;D